### DISCUSSION TITLE
tracee in Kubernetes

### DISCUSSION THREAD

**Discussion by itaysk** [aquasecurity/tracee#472] [Development]
tracee should be able to run in a Kubernetes. the goal is for a user to be able to start tracing one pod or container, without jumping through too many hoops.
this issue is for tracking Kubernetes related requests and tasks.

To be discussed:
- running tracee in pods vs running on nodes.
- what are the implications of running many instances of tracee on the same node?
- client/server architecture can be used to separate event collecting from filtering.
- we currently do detect containers, but we don't allow to target a specific container.
- consider filtering by mnt ns and cgroups

related to #20

**Comment by itaysk** [comment]
[bcc is starting to create a standard](https://github.com/iovisor/bcc/blob/master/docs/special_filtering.md) way for "container filtering" where by tools will implement options for mnt ns and cgroups filtering

**Comment by itaysk** [comment]
inspektor-gadget is a kubernetes native platform to integrate ebpf tools. there's a suggestion to add tracee as a tool, which will help us accelerate kubernetes integration: https://github.com/kinvolk/inspektor-gadget/issues/106

**Comment by ayoubeddafali** [comment]
Subscribed to thiis interesting thread
